File: sea-shell-service-spring-web-flux/src/main/java/org/jesperancinha/shell/webflux/data/SeaShellDto.java
package org.jesperancinha.shell.webflux.data;


import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import org.jesperancinha.shell.webflux.model.SeaShellLocation;
import org.jesperancinha.shell.webflux.model.ShellState;
import org.jesperancinha.shell.webflux.model.ShellType;

import java.beans.ConstructorProperties;
import java.math.BigDecimal;
import java.util.List;

@Builder
@Getter
@Setter
@EqualsAndHashCode
@ToString
@AllArgsConstructor
@NoArgsConstructor
public class SeaShellDto {
    private String name;
    private String scientificName;
    private List<SeaShellPersonDto> persons;
    private List<SeaShellCostumeDto> costumes;
}

File: sea-shell-service-spring-web-flux/src/main/kotlin/org/jesperancinha/shell/webflux/data/SeaShellDto.kt
package org.jesperancinha.shell.webflux.data

import lombok.Builder

@JvmRecord
data class SeaShellDto @Builder constructor(
    val name: String,
    val scientificName: String,
    val slogan: String,
    val persons: MutableList<SeaShellPersonDto?>,
    val costumes: MutableList<SeaShellCostumeDto?>,
    val personIds: List<Long>,
    val costumeIds: List<Long>
) {
    fun addPersons(persons: List<SeaShellPersonDto?>?) {
        this.persons.addAll(persons!!)
    }

    fun addCostumes(costumes: List<SeaShellCostumeDto?>?) {
        this.costumes.addAll(costumes!!)
    }
}